java/src/hu/holyoil/controller/SunController.java

```diff
@@ -54,13 +53,33 @@ public void RemoveAsteroid(Asteroid asteroid)  {
     }
 
     /**
-     * Amikor a turnsUntilNextSunstorm számláló eléri a 0-t elindít egy napvihart.
-     * <p>A pályán található összes aszteroida reagál rá.</p>
+     * Amikor a turnsUntilNextSunstorm számláló eléri a 0-t, ez a metódus hívodik meg.
+     * <p>Hatást gyakorol az öv aszteroidáinak nagyjából egyharmadára.</p>
      */
     public void StartSunstorm()  {
-       Logger.Log(this,"Starting sunstorm");
-       asteroids.forEach(Asteroid::ReactToSunstorm);
-       Logger.Return();
+        Logger.Log(this,"Starting sunstorm");
+
+        /* Collecting the asteroids to affect */
+        Set<Asteroid> chosenAsteroids = new HashSet<>();
+        Queue<Asteroid> traversingQueue = new LinkedList<>();
+
+        Random rand = new Random();
+        traversingQueue.add(
+                asteroids.get(
+                        rand.nextInt(asteroids.size())
+                )
+        );
+
+        while (chosenAsteroids.size() < asteroids.size() / 3 && !traversingQueue.isEmpty()) {
+           Asteroid currentAsteroid = traversingQueue.remove();
+           chosenAsteroids.addAll(currentAsteroid.GetNeighbours());
+           traversingQueue.addAll(currentAsteroid.GetNeighbours());
```

Comment:
I think lines 75 and 76 should be switched and traversingQueue should filter out asteroid that have already been chosen. Currently the starting asteroid may be checked many times.

Reply:
Yeah, just reworked it. But I'll write a test later for it to be bulletproof.